src/main/java/apollosupport/federation/EntityBatchDataFetcher.java

```diff
@@ -0,0 +1,15 @@
+package apollosupport.federation;
+
+import graphql.execution.DataFetcherResult;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import org.dataloader.BatchLoader;
+
+public class EntityBatchDataFetcher implements BatchLoader<DataFetchingEnvironment, DataFetcherResult<Object>>  {
```

Comment:
It might be possible to implement this in `GraphQLServiceBatchLoader` but will look into the later